.github/workflows/check-contracts.yml

```
@@ -1,48 +1,59 @@
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3  # Updated to v3
+
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@stable  # Better maintained action
+        with:
+          components: rust-src
+
+      - name: Cache cargo registry
+        uses: actions/cache@v3
```

actionlint (1.7.7)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)